internal/services/instance_types_service_test.go

```diff
@@ -0,0 +1,70 @@
+package services_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	clientStub "github.com/RHEnVision/provisioning-backend/internal/clients/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/dao/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+	"github.com/RHEnVision/provisioning-backend/internal/services"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListInstanceTypes(t *testing.T) {
+
+	t.Run("with region", func(t *testing.T) {
+		var names []string
+		ctx := stubs.WithAccountDaoOne(context.Background())
+		ctx = identity.WithTenant(t, ctx)
+		ctx = clientStub.WithSourcesClient(ctx)
+		ctx = clientStub.WithEC2Client(ctx)
+
+		req, err := http.NewRequestWithContext(ctx, "GET", "/api/provisioning/sources/1/instance_types?region=us-east-1", nil)
+		assert.Nil(t, err, fmt.Sprintf("Error creating a new request: %v", err))
```

Comment:
I disencourage from putting messages to assertions which are pretty obvious when you see them, but it's fine. If it was me, I would simply:

```suggestion
		assert.Nil(t, err)
```

This can be more easily copy-pasted.